subprojects/jacoco/src/testFixtures/groovy/org/gradle/testing/jacoco/plugins/fixtures/JacocoCoverage.groovy

```diff
@@ -27,7 +27,9 @@ final class JacocoCoverage {
 
     // Release notes: https://www.jacoco.org/jacoco/trunk/doc/changes.html
     static List<String> getSupportedVersionsByJdk() {
-        if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_15)) {
+        if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_17)) {
+            return filter(JacocoVersion.SUPPORTS_JDK_17)
+        } else if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_15)) {
```

Comment:
@bmuskalla I'm wondering why branch for Java 16 is missing here

```
if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_16))
  return filter(JacocoVersion.SUPPORTS_JDK_16)
```

where

```
JacocoVersion.SUPPORTS_JDK_16 = new JacocoVersion(0, 8, 6)
```

?

And thank you for adding missing pieces to my original changes 🤗 ❤️

Reply:
Thanks for catching, done.